### DISCUSSION TITLE
Issues, bugs and being stuck

### DISCUSSION THREAD

**Discussion by munoztd0** [munoztd0/DAP_2021#8] [General]
Friendly reminder for everyone to post their issues here (or in "issues" or anywhere). This is:

1) To share a problem and the solution to others
2) To save time, so I can work on fixing issues before I met you and have more time to explain/discuss more important things with you.

Thanks 

@lucilefavero @danielsgranja @ap1987-ap @faureunige

**Comment by lucilefavero** [comment]
Hi !

For the “Markdown and survival” exercise of this week, I set the model : 
```{r}
surv<-Surv(d$time,d$dead==1)
m1<-coxph(data=d,surv~ gvhd)
m1.diag<-cox.zph(m1)
```
And when I run these lines into the notebook, I have a warning but I am still able to make plot…
However, when I compile to have the pdf it raises the error : 

![Screenshot 2021-12-06 at 22 43 16](https://user-images.githubusercontent.com/54062914/144927700-5b34b5b2-dc5e-458f-a5ad-6e6e51ee95b6.png)

I have the same problem for all the other survival model I set. 

Thanks in advance for your response @munoztd0, 

Lucile

**Reply by munoztd0** [reply]
Could you upload the whole markdown file here or in your repo so I can have a look ?

**Reply by munoztd0** [reply]
PS: I have added/modified a few things on your Rmarkdown file such as to :
- Fix Margins 
- add an example of citations 
- put the image on the middle
You can see the changes I made by pulling the updated files on github (we can discuss about it next time we meet)

And I also added an "helper.R" file where I will the lines of code we discuss here for example its the function to plot assumptions of GLMs

**Reply by lucilefavero** [reply]
Thanks a lot ! And I upload the markdown in my folder.

**Comment by munoztd0** [comment]
As for your question, I checked your script and there is an issue with the model when you look at it:

![image](https://user-images.githubusercontent.com/43644805/145247556-e45db994-bcf6-4f1a-bf6a-c3666b0f9957.png)

As you see here, your model is empty..

It seems that the error came when you tried to subset the data 
![image](https://user-images.githubusercontent.com/43644805/145248010-2a6d6ec1-5364-4cc4-b401-5a320df15e2e.png)
Here you see that `d$dead==1` was false everywhere, instead `d$dead=="yes`" fixed the issue.

I made the changes to your files so don't forget to `git push` before (blue arrow).

**Reply by lucilefavero** [reply]
yes I see thanks a lot !

**Comment by munoztd0** [comment]
And for the overdispersion issue : 

I added that to your mardkown, let me know if you have more question

![image](https://user-images.githubusercontent.com/43644805/145253902-50f6b186-49d7-49df-99a2-120a70ff2994.png)
![image](https://user-images.githubusercontent.com/43644805/145253927-b0ad5a13-76d6-47a7-a289-197d41742ffd.png)

**Reply by lucilefavero** [reply]
It's ok, thanks a lot !

**Comment by danielsgranja** [comment]
Hello ! 

I hve a problem with the project sync on github. It says that "your branch is ahead of 'origin/main' by 9 commits.", and i cannot find how to come back from these commits, nor merge them with the main branch. When I try to pull, it works, but when i push, this appears.

<img width="505" alt="Capture d’écran 2021-12-10 123148" src="https://user-images.githubusercontent.com/94382469/145577539-d5022de5-b0c5-4a19-acc1-21e257cc62a6.png">

**Reply by munoztd0** [reply]
Hello, this is because you added some big file in your folder and github doesn't support that ..

That's not a big deal but it is just complicated to fix it in rstudio, either you can wait for monday and we will do it together or if you want to try from the command line (in your terminal not the R console) in the DAP_2021 folder: 
```
git rm --cached TinyTex.zip
git commit --amend -C HEAD
```

**Reply by danielsgranja** [reply]
Hi, I tryed and it didn't worked unfortunately.

**Reply by munoztd0** [reply]
Ok we will check tomorrow

**Reply by danielsgranja** [reply]
Hey, new problem with that ^^' 

When I tryed to pull, i got this
<img width="503" alt="Capture d’écran 2021-12-24 012051" src="https://user-images.githubusercontent.com/94382469/147301675-413a153f-db4e-410e-9f9d-1ddca10f04b0.png">

and after that i wasn't able to commit and push anymore with the same text as two weeks ago. I added manually the file i'm working on.

For that, I have difficulties in creating the Likert graph. I tryed anothed method to include the three emotions, but it doesn't work...

**Reply by munoztd0** [reply]
Let's see .. can we try to have a small zoom today a 2pm?

**Reply by danielsgranja** [reply]
Unfortunately today and tomorrow it will be difficult 😅

**Reply by munoztd0** [reply]
No problem! I check you DAP-pro2.rmd and I made a few changes to make likert work easier for you, if you can't pull form gtihub no worries just check it on github and keep your work advancing.

**Comment by danielsgranja** [comment]
Hello, I have a problem in comparing my models (Dap_pro.Rmd, line 106). I cannot use the anova to compare my models, so is there any formal way to compare them, or should I go with a text explanation on why one is better than the other?

**Reply by munoztd0** [reply]
Hello, 

I'm gonna check it out, but can you try to inverse the order of the models in the ANOVA function to see if it still doesn't work. But know that there is other ways to compare models. AIC is a good start (with extractAIC if it's not in the summary output),  the overall R², check if the degrees of freedom are "incoherent" , check the residuals plots (try sjPlot::plot_model(m, type="Diag") or go for the simple plot(m) ),  the ICC can also be a nice comparison but I will explain to you what it is later on but try (sjPlot::tab_model(m) )

PS: here is a cool introduction to Mixed models
http://www.john-ros.com/Rcourse/lme.html
Especially from 8.2.2
